### ISSUE TITLE
Literal enums

### ISSUE THREAD

**Issue by PanayotCankov** [Microsoft/TypeScript#10604]
We are using string enums a lot. The following example is what gets pretty close:
``` TypeScript
/**
 * Allowed directions.
 */
export declare type FlexDirection = "row" | "column";
export namespace FlexDirection {
    /**
     * Align items in a row.
     */
    export const ROW: "row" = "row";

    /**
     * Align items in a column.
     */
    export const COLUMN: "column" = "column";
}

declare function go(direction: FlexDirection); // Hover "FlexDirection to see docs.
go(FlexDirection.ROW); // Hover "ROW" to see docs.
go("row");
```

Is there a better way to add documentation to a string literal type? Sometimes the meaning of the values is not all that visible, or requires examples. The only thing I miss in a literal type such as `"row" | "column"` at the moment is I can not easily associate docs with the 2 values.

Does it sound feasible to add a shorthand syntax for this similar to:
``` TypeScript
/**
 * Allowed directions.
 */
export literal /* whatever the keywords */ enum FlexDirection {
    /**
     * Align items in a row.
     */
    ROW = "row",

    /**
     * Align items in a column.
     */
    COLUMN = "column"
}
```

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/3192

**Comment by kitsonk** [comment]
You mean #7642 right (of which #9407 resolved it)? #3192 was string literals.

**Comment by mhegazy** [comment]
my bad, it should be https://github.com/Microsoft/TypeScript/issues/1206.

#7642 does handle some of the cases, but there still a need for a string/symbol enum.
